### ISSUE TITLE
2 Questions for implementation

### ISSUE THREAD

**Issue by Zaquum** [google-research/growneuron#10]
First of all, thank you for sharing your surprising study with github. I am trying to quote your paper and have a few questions before I quote.

1. In your paper, you wrote that you use top-k left singular vector for estimating W^{new}_{l+1}. However, it seems that right singular vector is used in your code. Is it because the weight shape of tensorflow is transposed for input and output? 

2. Get back to your equation (11) in your paper. If a network have 3 neurons in 'l-1'th layer, 16 neurons in 'l'th layer and 32 neurons in 'l+1'th layer, which means the shape of weight for first linear layer is [16 x 3] and second is [32 x 16]. Then, the shape of expectation in equation (11) would be [32 x 3]. If I want to add the same number of neurons as the number of neurons that currently exist at 'l'th layer, which means adding 16 neurons to 'l'th layer. Then, I can't image how svd work for this case because we have only 3 eigenvalues thus we can't scale the top k left-singular vector with singular values . Also in that case, the svd operation will get errors for parameter 'k' in the function which must be between 1 and min(32, 3). Can you explain this case? 
 
3. In your models like vgg and wide resnet, the dropout layers are excluded. Can you explain why?

Thank you, Sincerely.

**Comment by evcu** [comment]
It has been a while I didn't look at the code. 

1) Only one of them would work shape-wise. So that must be it. 
2) You are right. max-3 neurons, though you can always train a bit and grow again. 
3) I don't think we had a specific reason. I don't think enabling drop-out is part of the architecture. I don't expect to get different results for dropout. Except one can potentially sample more diverse neurons. Might be a worthwhile research direction.
